### DISCUSSION TITLE
Boot RPI3B+ from USB drive ?

### DISCUSSION THREAD

**Discussion by ceejayemm** [rern/rAudio#160] [Fixed Issues]
Has anybody tried running rAudio-1 on an RPI3B+ booted from a USB flash drive rather than a SD card ?  According to the RaspberryPi website the 3B+ should be able to do this 'out of the box' - but has anybody tried with an rAudio-1 image ?  

The layout of my case means that in order to change the SD card on my RPi 3B+ I have to dissemble the case, take out the PI , change the card and then put it back together again - whereas the USB slots are easily accessible from outside the case and do not require the case to be taken apart.

Thanks for any feedback you might be able to provide.

Chris

**Comment by ceejayemm** [comment]
I think I might have answered my own question but...

I followed the instructions at https://github.com/rern/rOS to create a DIY installation of rAudio-1.  This went reasonably well in that I:

1. Used a Ubuntu PC to created the /BOOT and /ROOT partitions on a new 16GB USB flash drive 
2. Still on Ubuntu, opened a new Terminal session and ran the bash <( wget -qO - https://github.com/rern/rOS/raw/main/create-alarm.sh )  script to begin the installation process.
3. Answered the relevant questions from the script.
4. At the end (?) of the script it seemed to want search of the IP number of the PI (but I was at this point still on the Ubuntu PC) and then it wanted to reboot. Obviously I couldn't let this happen on the Ubuntu PC so unmounted the USB drive from the PC and transferred it back to the RPI PI3B+.
5. Booted the PI (from the USB drive) but rather than starting rAudio-1 it booted into an Arch-Linux user (root ?) account where the only thing available to me was a script file 'create-ros.sh'
6. Ran the script as './create-ros.sh' which appears to have kicked in the Arch-Linux system updates followed by the rAudio-1 installation. A little while later (more than the 15 mins suggested in the DIY installation web page) I was able to connect to the rAudio-1 web page and configure the installation as normal.

I seem to have Kernel: 5.10.11-4-ARCH armv7l and update 201201311 and I was able configure everything I needed except the Features page which errors with:

Errors: Unexpected token , in JSON at position 217
{ "autoplay" : false , "hostname" : "RAConservatory" , "lcd" : false , "login" : false , "mpdscribble" : false , "mpdscribbleval" : "^^" , "reboot" : "" , "snapserver" : false , "snapclient" : false , "snaplatency" : ▮, "snappassword" : "" , "streaming" : false , "localbrowser" : false , "localcursor" : false , "localrotate" : "NORMAL" , "localscreenoff" : 0 , "localzoom" : 1}

So I still can't access the RPI Browser setup to enable the Browser setting for this installation (noted previously #158) 

Everything else seems OK.

Did I complete the DIY installation correctly ? If not, what should I have done better.  Should I in fact have booted the RPI 3B+ from a Raspberry PI OS SD installation before running the DIY Installation process on the PI ?

 It might help if the installation process documentation has a simple list of the steps to carry out (and where to run them). The current set jump about a bit and have the several places where it appears to do steps already processed.

Chris

**Comment by ceejayemm** [comment]
@rern 
This problem still exists after update to 202102051.

BTW - I have now figured out what I was doing wrong with the ROS installer and this machine now has a fully working installation (apart from the problems with the Features page).

Chris

**Comment by ceejayemm** [comment]
@rern 
I think I have figured part of this problem out. I think its probably a ROS problem so I will raise an Issue on that site too.

I have a Pi Zero (standard image install) and a 3B+ (running the ROS DIY image) both running update 202102051. On the Zero the Features page works correctly but on the 3B+ I get the error noted above. After a bit of investigation I have found that (at least) the following files are missing from the 3B+

/etc/default/snapclient 
which on the Zero contains:
START_SNAPCLIENT=true
SNAPCLIENT_OPTS="--latency=800"

/etc/default/snapserver
which on the zero contains:
START_SNAPSERVER=true
SNAPSERVER_OPTS=""

I created both these files on the 3B+ and the Features page now at least renders and allows me to configure the RPi Browsers settings (which is all I wanted to do at this stage).  However when the page renders it appears a lot more 'bits' are missing, see the image below:

![image](https://user-images.githubusercontent.com/25418025/107142547-c3875780-6927-11eb-93fb-e7fedd383d71.png)

hence my also posting this info to the ROS site Issues page.

Chris

**Comment by kw-sudo** [comment]
Hi ceejayemm,

I know I'm running late, but is this still the only way for an USB-boot-option with the 3B+ modell? Even this RPi is USB-bootable out of the box? Or does it depend on the operating system? 

If I understood correctly, does it not work to simply install the recent r/audio image (SD-card) on an USB stick?

Regards,
Oliver

**Reply by ceejayemm** [reply]
@master-kw 
Since this ticket was raised I have found that I can now flash the USB drive directly with the current rAudio-1 release, using either Raspberry Pi Imager (Custom OS) or Balena Etcher.  There is no need to use the DIY installation these days.  It works well but does sometimes take a while to boot especially if I have the RPI 7in touch screen attached. The touch screen has other issues and I now tend to use the web app directly from another PC or tablet or the Android app to control this installation of rAudio-1.  I am currently up to date with the 20221123 updates applied.

I hope this helps.

Chris

**Reply by kw-sudo** [reply]
@ceejayemm 

Hi Chris,

thank you for quick reply. I've flashed an USB stick (SanDisk USB 3.1) with BalenaEtcher on my iMac and I don't use a touch screen or anything like that. I use the web gui for remote.

I had already read that the USB boot can take a while because the RPi has to register that the boot order has been changed and the USB controller is 'online'.

Do I simply have to retry the whole thing or do I have to pay attention to something special? Try a different USB stick? Or should that not matter?

Oliver

**Reply by ceejayemm** [reply]
@master-kw 
I don't do anything special other than place 'ssh' and 'wifi' files, configured to my needs on the 'boot' partition - note no extension to these filenames.  The 'ssh' file is blank and allows me to ssh into the RPi if necessary. The 'wifi' file should have your wifi connection information in it, like:

# Replace NAME and PASSWORD

Interface=wlan0
Connection=wireless
IP=dhcp
ESSID="Your Network SSID Name"
Key=Your Network Password
Security=wpa

I hope this helps.

Chris

**Reply by kw-sudo** [reply]
Thank you! My RPi is connected by wire so I should be fine with the entire informations.

Oliver

**Reply by kw-sudo** [reply]
I bought my RasPi used and the previous owner probably set the 'USB start bit' to off. I corrected that and now it works. 👍🏼 

Oliver
